### PR TITLE
CC Libraries - improvements

### PR DESCRIPTION
**Pull request by shaoshing**
this PR includes the following improvements over the CC Libraries:

* Memorize the last selected library, and restore it for new documents or reload - 1909
* Do not center library content vertically
* Can add graphic asset with long layer name - 2001
* Correctly render text style with no color - 1994
* Allow drag-n-drop on the initial library - 2008

@volfied please review

### CODE REVIEW

File: src/js/actions/libraries.js [hunk 1]
```diff
@@ -141,14 +149,14 @@ define(function (require, exports) {
                 representationType = _EXTENSION_TO_REPRESENTATION_MAP[fileExtension];
             }
         }
+        
+        var tempLayerName = (new Date().getTime()).toString();
```
Comment (shaoshing) [inline src/js/actions/libraries.js]: this is a walk-around for #2015

File: src/js/actions/libraries.js [hunk 2]
```diff
@@ -544,10 +552,18 @@ define(function (require, exports) {
      * @return {Promise}
      */
     var selectLibrary = function (id) {
-        return this.dispatchAsync(events.libraries.LIBRARY_SELECTED, { id: id });
+        var updatePreferencePromise = this.transfer(
+            preferencesActions.setPreference, _LAST_SELECTED_LIBRARY_ID_PREF, id);
+        
+        return Promise.resolve(updatePreferencePromise)
```
Comment (volfied) [inline src/js/actions/libraries.js]: I'd say dispatching the library selected event is higher priority than setting the preference here.

File: src/js/jsx/sections/libraries/assets/CharacterStyle.jsx [hunk 3]
```diff
@@ -88,10 +92,14 @@ define(function (require, exports, module) {
                 <div className="libraries__asset__desc">
                     <div>{font.name} {font.style}</div>
                     <div className="libraries__asset__desc-details">
-                        {fontSize.value}{fontSize.type}, {fontColorHex}
+                        {fontSize.value}{fontSize.type}
```
Comment (volfied) [inline src/js/jsx/sections/libraries/assets/CharacterStyle.jsx]: what if `fontSize` is null?

Comment (volfied) [inline src/js/jsx/sections/libraries/assets/CharacterStyle.jsx]: This happened to me with one of my older libraries, and I couldn't get things to work again.

**Comment by volfied** [comment]
Looks good, but while you're in there, if you can make CharacterStyle.jsx a bit more robust with some null checks, it would help.

Also needs rebase.

**Comment by shaoshing** [comment]
Added more checking when rendering Charstyle. Back to you @volfied

**Comment by volfied** [comment]
:+1:

**Comment by iwehrman** [comment]
Hey @shaoshing: if you stick a # in front of your bug numbers (e.g., in your PR summary), it will both link to that bug automatically and also create back-links from the issue to this PR. The former is useful in the short term, and the latter is often useful in the long term. #protip.

**Comment by shaoshing** [comment]
@iwehrman thanks for the tip :100:

**Pull request merged**